File: backend/servicio-inventario/src/main/java/com/regenta/inventario/aplicacion/CodigoResuelto.java
package com.regenta.inventario.aplicacion;

import java.math.BigDecimal;
import java.util.UUID;

/**
 * El producto al que apunta un código de barras, con su factor de conversión:
 * 1 para el código propio, N para un alterno (caja de N unidades).
 */
public record CodigoResuelto(
        UUID productoId,
        String sku,
        String nombre,
        BigDecimal factor,
        boolean esAlterno) {
}

File: app/packages/inventario/lib/src/datos/codigo_resuelto.dart
/// El producto al que apunta un código de barras, con su factor de conversión:
/// 1 para el código propio del producto, N para un alterno (caja de N).
class CodigoResuelto {
  const CodigoResuelto({
    required this.productoId,
    required this.sku,
    required this.nombre,
    required this.factor,
    required this.esAlterno,
  });

  final String productoId;
  final String sku;
  final String nombre;
  final num factor;
  final bool esAlterno;

  factory CodigoResuelto.desdeJson(Map<String, dynamic> json) => CodigoResuelto(
        productoId: json['productoId'] as String,
        sku: (json['sku'] ?? '') as String,
        nombre: (json['nombre'] ?? '') as String,
        factor: (json['factor'] ?? 1) as num,
        esAlterno: (json['esAlterno'] ?? false) as bool,
      );
}
